src/main/java/de/jamba/hudson/plugin/wsclean/PrePostClean.java

```diff
@@ -88,42 +94,62 @@ private void executeOnSlaves(AbstractBuild build, BuildListener listener) {
+		FilePath fp = null;
 		if (project instanceof TopLevelItem) {
-			FilePath fp = node.getWorkspaceFor((TopLevelItem) project);
-			if (fp != null) {
-				try {
-					fp.deleteContents();
-				} catch (IOException e) {
-					listener.getLogger().println(
-							"can't delete on node " + nodeName + "\n" + e.getMessage());
-					listener.getLogger().print(e);
-				} catch (InterruptedException e) {
-					listener.getLogger().println(
-							"can't delete on node " + nodeName + "\n" + e.getMessage());
-					listener.getLogger().print(e);
-				} catch (RequestAbortedException e){
-					listener.getLogger().println(
-							"can't delete on node " + nodeName + "\n" + e.getMessage());
-				}
-				
-			} else {
+			fp = node.getWorkspaceFor((TopLevelItem) project);
+		}
+		else if (project instanceof MatrixConfiguration) {
+			fp = getWorkspaceFor((MatrixConfiguration) project, listener, node, nodeName);
+		}
+		else {
+			listener.getLogger().println("Project is neither TopLevelItem nor MatrixConfiguration!? Cannot determine other workspaces!");
+		}
+		if (fp != null) {
+			try {
+				listener.getLogger().println("Deleting contents of " + fp);
+				fp.deleteContents();
+			} catch (IOException e) {
 				listener.getLogger().println(
-						"No workspace found on " + nodeName + ". Node is maybe offline.");
+						"can't delete on node " + nodeName + "\n" + e.getMessage());
+			} catch (InterruptedException e) {
+				listener.getLogger().println(
+						"can't delete on node " + nodeName + "\n" + e.getMessage());
+			} catch (RequestAbortedException e){
+				listener.getLogger().println(
+						"can't delete on node " + nodeName + "\n" + e.getMessage());
 			}
 		} else {
-			listener.getLogger().println("Project is no TopLevelItem!? Cannot determine other workspaces!");
+			listener.getLogger().println(
+					"No workspace found on " + nodeName + ". Node is maybe offline.");
+		}
+	}
+
+	private FilePath getWorkspaceFor(MatrixConfiguration matrixConfig, BuildListener listener, Node node, String nodeName) {
+		try {
+			String someWorkspacePath = matrixConfig.getSomeWorkspace().getRemote();
+			String matrixWorkspacePath = node.getWorkspaceFor((hudson.model.TopLevelItem) matrixConfig.getParent()).getRemote();
+			String[] parts = matrixWorkspacePath.split("[\\\\/]");
+			String[] configurationWorkspaceParts = someWorkspacePath.split(parts[parts.length - 2] + "[\\\\/]" + parts[parts.length - 1]);
+			String separator = (String) node.toComputer().getSystemProperties().get("file.separator");
+			String resultPath = matrixWorkspacePath + separator + configurationWorkspaceParts[configurationWorkspaceParts.length - 1];
+			return new FilePath(node.getChannel(), StringUtils.join(resultPath.split("[\\\\/]"), separator));
+		} catch (IOException e) {
+			listener.getLogger().println("Failure: " + nodeName + "\n" + e.getMessage());
+		} catch (InterruptedException e) {
```

Comment:
Not a good idea to catch and then merely _log_ an `InterruptedException` - if we're interrupted, it means we should abandon our cleanup ASAP as the user has asked for the build to be aborted (or similar issues).
In general, not catching the exception and allowing it to escape the method is the easiest option.